docs/text-predicates.rst

```diff
@@ -0,0 +1,215 @@
+==========================
+JanusGraph Text Predicates
+==========================
+
+JanusGraph provides APIs to query the data using Text Predicates. The types of Text Predicates available
+for querying depends on the Indexing backend used.
+Hence, be careful with using the Text Predicates, and refer to original JanusGraph docs_.
+
+.. _docs: https://docs.janusgraph.org/latest/
+
+
+----------------------------
+Available Text Predicates
+----------------------------
+
+The Text Predicates are broadly divided into 2 categories as:
+    - Text search predicates which match against the individual words inside a text string after it has been tokenized.
+        These predicates are not case sensitive.
+
+        - TextContains_
+        - TextContainsFuzzy_
+        - TextContainsPrefix_
+        - TextContainsRegex_
+
+    - String search predicates which match against the entire string value
+
+        - TextFuzzy_
+        - TextPrefix_
+        - TextRegex_
+
+.. note::
```

Comment:
I don't know `.rst` but shouldn't this be formatted else `**NOTE**` like you did elsewhere?